test/common/perf_monitor_test.cpp

```diff
@@ -2,92 +2,71 @@
 
 #include <thread>  //NOLINT
 
-#include "common/macros.h"
 #include "common/managed_pointer.h"
-#include "common/scoped_timer.h"
 #include "main/db_main.h"
-#include "test_util/storage_test_util.h"
 #include "test_util/test_harness.h"
 
 namespace terrier {
 
 class PerfMonitorTests : public TerrierTest {
- public:
-  static void CreateAndDestroyCatalog(common::PerfMonitor::PerfCounters *const counters) {
-    common::PerfMonitor monitor(false);
+};
 
-    auto db_main = terrier::DBMain::Builder().SetUseGC(true).SetUseCatalog(true).Build();
-    monitor.Start();
-    db_main.reset();
+template <bool inherit, typename perf_counters>
+static void CreateAndDestroyCatalog(perf_counters *const counters) {
+  common::PerfMonitor<inherit> monitor;
+  monitor.Start();
 
-    monitor.Stop();
-    *counters = monitor.Counters();
-  }
+  auto db_main = terrier::DBMain::Builder().SetUseGC(true).SetUseCatalog(true).Build();
+  db_main.reset();
 
-  static void JustSleep(common::PerfMonitor::PerfCounters *const counters) {
-    common::PerfMonitor monitor(false);
-    monitor.Start();
-    std::this_thread::sleep_for(std::chrono::seconds(2));
-    monitor.Stop();
-    *counters = monitor.Counters();
-  }
-};
+  monitor.Stop();
+  *counters = monitor.Counters();
+}
+
+template <bool inherit, typename perf_counters>
+static void JustSleep(perf_counters *const counters) {
+  common::PerfMonitor<inherit> monitor;
+  monitor.Start();
+
+  std::this_thread::sleep_for(std::chrono::seconds(2));
+
+  monitor.Stop();
+  *counters = monitor.Counters();
+}
 
 /**
  * Simple test that spins off 2 threads. One thread builds and then destroys a Catalog. The other sleeps. The parent
- * waits. We then do comparisons on their respective perf counters based on the work performed.
+ * waits. We can't make strong assertions about their values due to scheduling uncertainty.
  */
-// NOLINTNEXTLINE
-TEST_F(PerfMonitorTests, BasicTest) {
-  common::PerfMonitor parent_monitor(false);
-  common::PerfMonitor::PerfCounters parent_counters, catalog_counters, sleep_counters;
+template <bool inherit, typename perf_counters>
+static void UnbalancedChildrenThreads() {
+  common::PerfMonitor<inherit> parent_monitor;
+  perf_counters parent_counters, catalog_counters, sleep_counters;
   parent_monitor.Start();
-  std::thread thread1(CreateAndDestroyCatalog, &catalog_counters);
-  std::thread thread2(JustSleep, &sleep_counters);
+  std::thread thread1(CreateAndDestroyCatalog<inherit, perf_counters>, &catalog_counters);
+  std::thread thread2(JustSleep<inherit, perf_counters>, &sleep_counters);
   thread1.join();
   thread2.join();
   parent_monitor.Stop();
   parent_counters = parent_monitor.Counters();
```

Comment:
Does this test now essentially just start and stop the monitor?

Happy to LGTM the rest otherwise.

Reply:
The tests are hard to evaluate deterministically since depending on your kernel flags, perf counters might be 0. On a Mac they'll definitely be 0. I have tests there as a sanity check for compilation and to also have a quick test to toss a breakpoint in on a system to verify the counters are working correctly.